### ISSUE TITLE
Mask prediction

### ISSUE THREAD

**Issue by Steve-Tod** [microsoft/GLIP#28]
Thank you for your great work! I tried the demo and it's insanely well!

I'm wondering if your model contains a mask prediction head. Because in the `GLIPDemo` there's a `show_mask_heatmaps` parameter. When I set it to true, the prediction does not has `mask` field and therefore failed. 

Do you have pretrained model with a mask prediction head?

**Comment by Haotian-Zhang** [comment]
@Steve-Tod Thank you for the question! This is within expectation since our GLIPv1 model is only focusing on object detection and phrase grounding. However, our recent GLIPv2 paper do have the capibility with instance segmentation and referring image segmentation. Please refer to the Appendix of v2 paper for further details. The codes and models are under internal reviews and will be published soon. Please stay tuned, thanks!

**Comment by Steve-Tod** [comment]
Great, looking forward to that!

**Comment by Steve-Tod** [comment]
Just a short reminder, the Fig 1 of your paper seems to show the wrong prompt for the first two pictures.
![image](https://user-images.githubusercontent.com/26958167/179033986-1a6977f5-d048-4b2b-ae29-a5522715c667.png)

**Comment by Haotian-Zhang** [comment]
@Steve-Tod This is not wrong since the prompt is a concatenation of detection categories (We only list a few). The apple is one of the categories in first pic, so are the bus, windowshield, mirror in second pic.

**Comment by Steve-Tod** [comment]
Oh I see, thank you for clarification!

**Comment by Haotian-Zhang** [comment]
No problem!

**Comment by Steve-Tod** [comment]
@Haotian-Zhang  Sorry to bother again, but do you have an expectation on when the GLIP2 will be released? I really need the segmentation and if it's not gonna be released in the short future I'm thinking about applying MaskRCNN to GLIP prediction.

**Comment by Haotian-Zhang** [comment]
@Steve-Tod We are going to release the codes when the v2 paper review is done, so it still needs some time. In this case, I have a quicker and easier suggestion to extend v1 to mask prediction: You can just use some backbone (HRNet) to extract the features from the proposals by v1, and then add the mask prediction heads to predict binary masks. The pixel categories of the binary masks can be easily replaced by the classification results from v1. Hope that helps!

**Comment by Colin97** [comment]
Hi @Haotian-Zhang, what do you mean by "The pixel categories of the binary masks can be easily replaced by the classification results from v1." Do we need the ground truth segmentation mask to train the binary mask prediction head?

**Comment by Haotian-Zhang** [comment]
@Colin97 For the binary mask, you only need to predict whether this pixel belongs to the object or not. In this case, the class of certain pixels can only be assigned by the class of its proposals.

**Comment by Colin97** [comment]
But how to supervise the training of the segmentation head? Treat all pixels within the proposed bbox (by GLIP) as positive? Or use additional ground truth masks as supervision?

**Comment by Haotian-Zhang** [comment]
@Colin97 The training definitely needs the ground truth mask as the supervision. This aforementioned approach is to deal with the classification of the pixels in a quick way. However, in GLIPv2, we are predicting the mask in a "grounding" style, and you can refer to the paper for more details.

**Comment by Colin97** [comment]
Thank you! @Haotian-Zhang
